axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderInitValueServiceImpl.java

```diff
@@ -0,0 +1,161 @@
+package com.axelor.apps.sale.service.saleorder;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.base.db.PrintingSettings;
+import com.axelor.apps.base.db.repo.CompanyRepository;
+import com.axelor.apps.base.service.BankDetailsService;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.base.service.user.UserService;
+import com.axelor.apps.sale.db.SaleOrder;
+import com.axelor.apps.sale.db.repo.SaleOrderRepository;
+import com.axelor.apps.sale.service.app.AppSaleService;
+import com.axelor.auth.AuthUtils;
+import com.axelor.auth.db.User;
+import com.axelor.db.mapper.Mapper;
+import com.axelor.studio.db.AppBase;
+import com.axelor.studio.db.AppSale;
+import com.google.inject.Inject;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SaleOrderInitValueServiceImpl implements SaleOrderInitValueService {
+
+  protected AppBaseService appBaseService;
+  protected CompanyRepository companyRepository;
+  protected AppSaleService appSaleService;
+  protected UserService userService;
+  protected BankDetailsService bankDetailsService;
+
+  @Inject
+  public SaleOrderInitValueServiceImpl(
+      AppBaseService appBaseService,
+      CompanyRepository companyRepository,
+      AppSaleService appSaleService,
+      UserService userService,
+      BankDetailsService bankDetailsService) {
+    this.appBaseService = appBaseService;
+    this.companyRepository = companyRepository;
+    this.appSaleService = appSaleService;
+    this.userService = userService;
+    this.bankDetailsService = bankDetailsService;
+  }
+
+  @Override
+  public Map<String, Object> getOnNewInitValues(SaleOrder saleOrder) throws AxelorException {
+    Map<String, Object> initValues = new HashMap<>();
+    initValues.putAll(saleOrderDefaultValues(saleOrder));
+    initValues.putAll(getInAti(saleOrder));
+    initValues.putAll(getBankDetails(saleOrder));
+    initValues.putAll(getGroupProductsOnPrintings(saleOrder));
+    return initValues;
+  }
+
+  protected Map<String, Object> saleOrderDefaultValues(SaleOrder saleOrder) {
+    Map<String, Object> saleOrderMap = saleOrderDefaultValuesMap(saleOrder);
+    for (Map.Entry<String, Object> entry : saleOrderMap.entrySet()) {
+      Mapper.of(SaleOrder.class).set(saleOrder, entry.getKey(), entry.getValue());
+    }
```

Comment:
You could even move this code at the end of `getOnNewInitValues` since it is the only public method, or even define a protected `getOnNewInitValuesMap` which will only return the map